aptos-move/framework/aptos-framework/sources/account.move

```diff
@@ -798,12 +804,21 @@ module aptos_framework::account {
 
     public(friend) fun register_coin<CoinType>(account_addr: address) acquires Account {
         let account = borrow_global_mut<Account>(account_addr);
-        event::emit_event<CoinRegisterEvent>(
-            &mut account.coin_register_events,
-            CoinRegisterEvent {
-                type_info: type_info::type_of<CoinType>(),
-            },
-        );
+        if (std::features::module_event_migration_enabled()) {
+            event::emit(
+                CoinRegister {
+                    account: account_addr,
+                    type_info: type_info::type_of<CoinType>(),
+                },
+            );
+        } else {
```

Comment:
This is a good example of avoiding double-emission. I believe the migration of the other core V2 events should follow a similar approach.